src/nanoarrow/nanoarrow.h

```diff
@@ -276,9 +277,18 @@ enum ArrowTimeUnit {
 /// \brief Initialize the fields of a schema
 ///
 /// Initializes the fields and release callback of schema_out.
-ArrowErrorCode ArrowSchemaInit(struct ArrowSchema* schema_out);
-
-/// \brief Make a (full) copy of a schema
+ArrowErrorCode ArrowSchemaInit(struct ArrowSchema* schema, enum ArrowType type);
+
+ArrowErrorCode ArrowSchemaSetFixedSize(struct ArrowSchema* schema, int32_t fixed_size);
+ArrowErrorCode ArrowSchemaSetDecimalPrecision(struct ArrowSchema* schema,
+                                              int32_t decimal_precision,
+                                              int32_t decimal_scale);
+ArrowErrorCode ArrowSchemaSetTimeUnit(struct ArrowSchema* schema,
+                                      enum ArrowTimeUnit time_unit);
+ArrowErrorCode ArrowSchemaSetTimezone(struct ArrowSchema* schema,
+                                      struct ArrowStringView* timezone);
```

Comment:
Perhaps an API using C99 designated initializers would be more user-friendly? How about being able to write e.g.:
```c
ArrowSchemaInit(
  &schema1,
  {.type = NANOARROW_TYPE_FIXED_SIZE_BINARY, .fixed_size = 3});
ArrowSchemaInit(
  &schema2,
  {.type = NANOARROW_TYPE_DECIMAL128,
   .decimal_precision = 9, .decimal_scale = 2});
ArrowSchemaInit(
  &schema3,
  {.type = NANOARROW_TYPE_DURATION,
   .time_unit = NANOARROW_TIME_UNIT_NANO,
   .nullable = false});
```

Reply:
The API could look like:
```c
struct ArrowSchemaInitializer {
  ArrowType type;
  _Bool nullable;
  ArrowTimeUnit time_unit;
  int32_t decimal_precision;
  // etc.
};

ArrowErrorCode ArrowSchemaInit(struct ArrowSchema*, struct ArrowSchemaInitializer);
```

Reply:
We can also provide macros to ensure people pass the right fields:
```c
ArrowSchemaInit(&schema1, ARROW_SCHEMA_FIXED_SIZE_BINARY(3));
ArrowSchemaInit(&schema2, ARROW_SCHEMA_DECIMAL128(9, 2));
ArrowSchemaInit(&schema3, ARROW_SCHEMA_DURATION(NANO));
```